### PR TITLE
Feature/update kass configs

### PR DESCRIPTION
**Pull request by pslocum**
These changes include a minor modification to the upgrades in which Kassiopeia field solutions are moved from the Kassiopeia scope to the KEMField scope.   While the field solutions remain in the KEMField scope, their superposition is relocated to the Kassiopeia scope.  This appears to stop the memory leak reported in https://github.com/project8/locust_mc/issues/220 .

### CODE REVIEW

**Comment by renereimann** [comment]
Hej Penny, thanks for this pull request. I just looked at it briefly but it seems to be good. 
I did not check it for memory leaking yet. 
In principle I would guess this is an Kassiopeia issue and should be fixed in kassiopeia but for the time till this is fixed in kassiopeia the changes seem to be good. I will test the changes in the next days.

**Comment by renereimann** [comment]
Hej @pslocum,
I looked through the Kassiopeia implementation to check if I find a problem there. The difference between the "two" different methods is that the "SuperpositionField" has an option "use_caching" that caches the field for already computed positions. The KRootMagneticField which is set in the "Simulation" tag does not have caching implemented. 
In the config files currently the SuperpositionField has "use_caching=true" always. This has the effect that the "caching" map increases with runtime, if the positions do not repeat. If using the "KRootMagneticField" by adding fields in the simulation tag does not cache but always loop through the individual fields and recalculates them. The "memory leak" like behaviour thus is the effect from using caching. I propose that we simply turn of caching instead of moving the individual fields into the simulation tag. This allows to make the magnetic field more modular. The XML-Kassiopeia syntax does not allow to comment out an attribute of a tag but in the KEMField / KMagneticSuperpositionField the fields are added with an individual tag. They can be implemented configurable.

File: Config/FreeSpaceGeometry_V00_00_10.xml [hunk 1]
```diff
@@ -189,7 +189,7 @@
         use_caching="true"
```
Comment (renereimann) [inline Config/FreeSpaceGeometry_V00_00_10.xml:189]: Change to false

File: Config/FreeSpaceGeometry_V00_00_11_1GHz.xml [hunk 2]
```diff
@@ -189,7 +189,7 @@
         use_caching="true"
```
Comment (renereimann) [inline Config/FreeSpaceGeometry_V00_00_11_1GHz.xml:189]: Set to false

File: Config/JustKass.xml.in [hunk 3]
```diff
@@ -461,7 +461,8 @@
         run="1"
         seed="[seed]"
         events="10"
-        magnetic_field="total_magnetic_field"
```
Comment (renereimann) [inline Config/JustKass.xml.in:464]: Revert to previous implementation

File: Config/JustKassFieldMap.xml.in [hunk 4]
```diff
@@ -457,7 +457,8 @@
         run="1"
         seed="[seed]"
         events="1"
-        magnetic_field="total_magnetic_field"
```
Comment (renereimann) [inline Config/JustKassFieldMap.xml.in:460]: Revert to previous implementation

File: Config/LocustKass_Cavity_Template.xml.in [hunk 5]
```diff
@@ -449,7 +449,8 @@
         run="1"
         seed="[seed]"
         events="1"
-        magnetic_field="total_magnetic_field"
```
Comment (renereimann) [inline Config/LocustKass_Cavity_Template.xml.in:452]: Revert to previous implementation

File: Config/LocustKass_EGun_Template.xml.in [hunk 6]
```diff
@@ -449,7 +449,8 @@
         run="1"
         seed="[seed]"
         events="1"
-        magnetic_field="total_magnetic_field"
```
Comment (renereimann) [inline Config/LocustKass_EGun_Template.xml.in:452]: Revert to previous implementation.

File: Config/LocustKass_FreeSpace_Template.xml.in [hunk 7]
```diff
@@ -443,7 +443,8 @@
         run="1"
         seed="[seed]"
         events="3"
-        magnetic_field="total_magnetic_field"
```
Comment (renereimann) [inline Config/LocustKass_FreeSpace_Template.xml.in:446]: Revert to previous implementation.

File: Config/LocustKass_Waveguide_Template.xml.in [hunk 8]
```diff
@@ -457,7 +457,8 @@
         run="1"
         seed="[seed]"
         events="3"
-        magnetic_field="total_magnetic_field"
```
Comment (renereimann) [inline Config/LocustKass_Waveguide_Template.xml.in:460]: Revert to previous implementation.

File: Config/WaveguideGeometry.xml [hunk 9]
```diff
@@ -175,7 +175,7 @@
         use_caching="true"
```
Comment (renereimann) [inline Config/WaveguideGeometry.xml:175]: change to false

**Review by renereimann — CHANGES_REQUESTED** [review]
As mentioned in the previous comment I am strongly in favour to keep the field configuration modular from the simulation configuration. The memory leak like behaviour is a result from using caching which can be turned of in the KMagneticSuperpositionField configuration itself. I like the renaming of the electromagnet_field to trap_field as it is more specific and self-explaining for our purpose.

**Review by pslocum — COMMENTED** [review]
After testing, these changes appear to fix the memory usage problem.

**Review by renereimann — APPROVED** [review]

**Pull request merged**